File: server/src/main/java/com/grpc/empuje_comunitario/application/user/EncrypterImpl.java
package com.grpc.empuje_comunitario.application.user;

import com.grpc.empuje_comunitario.infrastructure.security.BCryptPasswordEncryptor;
import org.springframework.stereotype.Component;

@Component
public class EncrypterImpl implements Encrypter {

    private final BCryptPasswordEncryptor encryptor;

    public EncrypterImpl(
            BCryptPasswordEncryptor encryptor
    ) {
        this.encryptor = encryptor;
    }

    @Override
    public String encrypt(String data) {
        return encryptor.encrypt(data);
    }
}

File: server/src/main/java/com/grpc/empuje_comunitario/application/user/EncrypterImpl.kt
package com.grpc.empuje_comunitario.application.user

import com.grpc.empuje_comunitario.infrastructure.security.BCryptPasswordEncryptor
import org.springframework.stereotype.Component

@Component
class EncrypterImpl(
    private val encryptor: BCryptPasswordEncryptor
) : Encrypter {
    override fun encrypt(data: String): String {
        return encryptor.encrypt(data)
    }
}